### PR TITLE
Add pathway analysis muddiest points notebook

### PR DESCRIPTION
**Pull request by jaclyn-taroni**
I am in appointments and meetings most of tomorrow morning, so I decided I would put this together based on the information we have so far. This supplemental notebook covers:

* Saving GSEA plots for all pathways/gene sets returned as significant 2 ways (`purrr` and `for` loop)
* How distributions will look different depending on a gene set size and how that impacts significance

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
HTML preview version for the reviewer's convenience: https://htmlpreview.github.io/?https://github.com/AlexsLemonade/2021-june-training/blob/78effa6248e83ea446e9a373644f37cd3f6f5191/supplemental_notebooks/2021-07-01_muddiest_points.nb.html

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+```
+
+Let's create a ridgeline plot of enrichment scores.
+
+```{r}
+random_es %>%
+  ggplot2::ggplot(ggplot2::aes(enrichmentScore,
+                               pathway_size,
+                               fill = pathway_size)) +
+  ggridges::geom_density_ridges() +
```
Comment (jashapiro) [inline supplemental_notebooks/2021-07-01_muddiest_points.Rmd:246]: We don't have `ggridges`! We can show how to install it here (or a bit earlier) if we like.

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+First, let's make sure we have a `plots/hodgkins/gsea` directory to hold our output.
+
+```{r}
+plots_dir <- file.path("plots", "hodgkins", "gsea")
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+First, let's look at (one) way to save each of the significant pathways using `purrr`.
```
Comment (jashapiro) [inline supplemental_notebooks/2021-07-01_muddiest_points.Rmd:118]: Only get to be first once!
```suggestion
Now, let's look at (one) way to save each of the significant pathways using `purrr`.
```

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+# Now we need to "walk" through both the plots and the gene set names
+plot_list %>%
+  purrr::walk2(as.list(significant_gene_sets),  # contains pathway names
+               ~ ggplot2::ggsave(plot = .x,   # plot itself (first list)
+                                 filename = file.path(plots_dir, 
+                                                      paste0(.y, # pathway name (second list)
+                                                             "_gsea_plot.png")),
+                                 width = 11, height = 7
+               ))
```
Comment (jashapiro) [inline supplemental_notebooks/2021-07-01_muddiest_points.Rmd:138]:
```suggestion
purrr::walk2(
  plot_list, # the list of plots
  as.list(significant_gene_sets),  # the list of pathway names
  ~ ggplot2::ggsave( # the function we are applying
      plot = .x,   # plot itself (from the first list)
      filename = file.path(plots_dir, 
                           paste0(.y, # pathway name (from the second list)
                                  "_gsea_plot.png")),
      width = 11, 
      height = 7
    )
)
```

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+for (pathway in significant_gene_sets) {
+  # Create the plot and assign it to `p`
+  p <- enrichplot::gseaplot(gsea_results,
+                            geneSetID = pathway,  # Pathway name
+                            title = pathway) # Pathway name
+  # Save the plot, using the pathway name as part of the filename
+  # Note that this will write over the files we saved with the purrr method!
+  ggplot2::ggsave(plot = p, filename = file.path(plots_dir,
+                                                 paste0(pathway, "_gsea_plot.png")),
+                  width = 11, height = 7)
```
Comment (jashapiro) [inline supplemental_notebooks/2021-07-01_muddiest_points.Rmd:153]: Is this reformat necessary? No, but I think I like it better.
```suggestion
  ggplot2::ggsave(plot = p, 
                  filename = file.path(plots_dir, paste0(pathway, "_gsea_plot.png")),
                  width = 11, height = 7)
```

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+## Libraries
+
+```{r}
+library(magrittr)
+# Package to run GSEA
+library(clusterProfiler)
+# Package that contains the MSigDB gene sets in tidy format
+library(msigdbr)
+```
+
```
Comment (jashapiro) [inline supplemental_notebooks/2021-07-01_muddiest_points.Rmd:43]: Adding a chunk to install ggridges
~~~suggestion

For some later plots, we are going to use the `ggridges` package, which may not be installed, so let's install it here (only if it is not already installed):

```{r}
if (! "ggridges" %in% installed.packages()){
  install.package("ggridges")
}
```
~~~

**Review by jashapiro — COMMENTED** [review]
This looks good to me, with a couple of small additions & changes. But I will say that I am not sure I am the best judge right now. I might like somebody else to take a look as well!

File: supplemental_notebooks/2021-07-01_muddiest_points.Rmd (hunk 5 above)
Comment (jaclyn-taroni) [reply supplemental_notebooks/2021-07-01_muddiest_points.Rmd:43]: I guess I must have installed it myself previously!

**Review by jashapiro — APPROVED** [review]
Approving in case you want to merge (after rerendering html)

**Pull request merged**